demos/common/processing/async_pipeline.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+// Copyright (C) 2018-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "async_pipeline.h"
+#include <samples/args_helper.hpp>
+#include <cldnn/cldnn_config.hpp>
+#include <samples/slog.hpp>
+
+using namespace InferenceEngine;
+
+PipelineBase::PipelineBase(ModelBase* modelInstance, const CnnConfig& cnnConfig, InferenceEngine::Core* engine){
+    model.reset(modelInstance);
+
+    auto ie = engine ?
+        std::unique_ptr<InferenceEngine::Core, void(*)(InferenceEngine::Core*)>(engine, [](InferenceEngine::Core*) {}) :
+        std::unique_ptr<InferenceEngine::Core, void(*)(InferenceEngine::Core*)>(new InferenceEngine::Core, [](InferenceEngine::Core* ptr) {delete ptr; });
```

Comment:
Is this really necessary? I could only find one use of `PipelineBase` in the code, so we clearly don't need both of these options.

Reply:
Support for external ie was aded by request from @eizamaliev , I suggest to keep both apporaches

Reply:
But why? I think we only need the external approach.

Reply:
internal approch is better for simple solutions as we hide IeCore from application level code (no need to initialize it, etc...). So I'd like to keep both approached

Reply:
for simple solution is better to have one approach. Having a choice is always complicates the things

Reply:
done